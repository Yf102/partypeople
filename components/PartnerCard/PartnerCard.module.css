.light {
    background: var(--card-color-light);
}

.paragraph {
    color: var(--paragraph-text-color);
}