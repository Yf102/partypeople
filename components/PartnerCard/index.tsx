import React from 'react'
import { PartnerCardType } from 'services/types/partners'
import styles from './PartnerCard.module.css'
import cn from 'classnames'
import Icon from 'components/Icon'

const PartnerCard = (p: PartnerCardType) => {
  return (
    <div
      data-testid='partner-card-element'
      className={cn(
        styles['light'],
        'col-span-12 col-start-1 text-center block rounded-lg px-8 py-10 mb-4'
      )}
    >
      <Icon className='opacity-30' width={42} height={42} icon={'icon_user'} />
      <div className={styles.paragraph}>{p.id}</div>
      <h1 className='text-xl'>{p.name}</h1>
      <div className={cn(styles.paragraph, 'mb-0')}>
        {Number(p.distance.toFixed(3))} km
      </div>
    </div>
  )
}

export default PartnerCard
