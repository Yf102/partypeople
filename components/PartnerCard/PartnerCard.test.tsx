import React from 'react'
import { render, screen } from 'tests/test-utils'

import PartnerCard from 'components/PartnerCard/index'
import { SinglePartner } from 'tests/__mocks__/data/Partners'

describe('PartnerCard', () => {
  it('It renders PartnerCard', async () => {
    const { asFragment } = render(
      <PartnerCard
        id={SinglePartner.id}
        name={SinglePartner.name}
        distance={SinglePartner.distance}
      />
    )
    expect(screen.getByTestId('partner-card-element')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})
