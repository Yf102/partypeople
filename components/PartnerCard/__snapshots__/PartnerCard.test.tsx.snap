// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerCard It renders PartnerCard 1`] = `
<DocumentFragment>
  <div
    class="light col-span-12 col-start-1 text-center block rounded-lg px-8 py-10 mb-4"
    data-testid="partner-card-element"
  >
    <img
      alt=""
      class="opacity-30"
      data-nimg="future"
      decoding="async"
      height="42"
      src="/icons/icon_user.svg"
      srcset="/icons/icon_user.svg 1x, /icons/icon_user.svg 2x"
      style="color: transparent;"
      width="42"
    />
    <div
      class="paragraph"
    >
      1
    </div>
    <h1
      class="text-xl"
    >
      Jamelia Waller
    </h1>
    <div
      class="paragraph mb-0"
    >
      11.732 km
    </div>
  </div>
</DocumentFragment>
`;
